process/elasticProcessor.go

```diff
@@ -541,20 +541,33 @@ func (ei *elasticProcessor) SaveRoundsInfo(info []*data.RoundInfo) error {
 	return ei.elasticClient.DoBulkRequest(buff, elasticIndexer.RoundsIndex)
 }
 
-func (ei *elasticProcessor) indexAlteredAccounts(timestamp uint64, alteredAccounts data.AlteredAccountsHandler) error {
+func (ei *elasticProcessor) indexAlteredAccounts(
+	timestamp uint64,
+	alteredAccounts data.AlteredAccountsHandler,
+	pendingBalances map[string]*data.AccountInfo,
+) error {
 	regularAccountsToIndex, accountsToIndexESDT := ei.accountsProc.GetAccounts(alteredAccounts)
 
 	err := ei.SaveAccounts(timestamp, regularAccountsToIndex)
 	if err != nil {
 		return err
 	}
 
-	return ei.saveAccountsESDT(timestamp, accountsToIndexESDT)
+	return ei.saveAccountsESDT(timestamp, accountsToIndexESDT, pendingBalances)
 }
 
-func (ei *elasticProcessor) saveAccountsESDT(timestamp uint64, wrappedAccounts []*data.AccountESDT) error {
+func (ei *elasticProcessor) saveAccountsESDT(
+	timestamp uint64,
+	wrappedAccounts []*data.AccountESDT,
+	pendingBalances map[string]*data.AccountInfo,
+) error {
 	accountsESDTMap := ei.accountsProc.PrepareAccountsMapESDT(wrappedAccounts)
-	err := ei.indexAccountsESDT(accountsESDTMap)
+	// merge maps
```

Comment:
can remove the comment and extract the for-loop in a function called: `mergeMaps`

Reply:
done added a method.